pkg/jaeger/jaeger_event.go

```diff
@@ -0,0 +1,98 @@
+package jaeger
+
+import (
+	"strconv"
+	"time"
+
+	jaegerModel "github.com/jaegertracing/jaeger/model"
+
+	"github.com/pavolloffay/jaeger-local-storage-poc/pkg/model"
+)
+
+// Event The jaeger span version of the event
+type Event struct {
+	tenant    string
+	timestamp time.Time
+	fields    []*model.Field
+}
+
+// NewEvent creates an event from a Jaeger span
+func NewEvent(span *jaegerModel.Span) model.Event {
+	fields := make([]*model.Field, 0)
+	fields = append(fields, model.NewField("traceId", model.StringType, []byte(span.TraceID.String())))
+	fields = append(fields, model.NewField("spanId", model.StringType, []byte(span.SpanID.String())))
+	fields = append(fields, model.NewField("service", model.StringType, []byte(span.Process.ServiceName)))
+	fields = append(fields, model.NewField("operation", model.StringType, []byte(span.OperationName)))
+	fields = append(fields, model.NewField("duration", model.Int64Type, []byte(strconv.FormatInt(span.Duration.Nanoseconds(), 10))))
+
+	// Iterate through process tags - intercept tenant separately
+	tenant := ""
+
+	for _, kv := range span.Process.Tags {
+		if kv.GetKey() == "tenant" {
+			tenant = kv.GetVStr()
+			continue
+		}
+		if field := convertTag(&kv); field != nil {
```

Comment:
Would it useful to differentiate between a span field from a tag? What happens nowadays if a span has a tag named `service`, or `spanId`, or `traceId`? In this code, the tags will override the fields, right?

Reply:
Agree, will differentiate the names